src/internal/task/task.proto

```diff
@@ -13,20 +13,18 @@ enum State {
   FAILURE = 3;
 }
 
+message Group {}
+
 message Task {
   string id = 1 [(gogoproto.customname) = "ID"];
-  google.protobuf.Any data = 2;
-}
-
-message TaskInfo {
-  Task task = 1;
   State state = 2;
```

Comment:
These fields, including state could potentially be a oneof for RUNNING, SUCCESS, and FAILURE.